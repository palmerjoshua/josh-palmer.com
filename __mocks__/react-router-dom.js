import React, { Component } from 'react';

const rrd = require('react-router-dom');

class MockLink extends Component {
    render() {
        return (
            <a href={this.props.to} style={{display: 'inline-block', height: '100%', width: '90%', padding: 10}}>{this.props.children}</a>
        );
    }
}

rrd.BrowserRouter = ({children}) => <div>{children}</div>;
rrd.Link = MockLink;
module.exports = rrd;
