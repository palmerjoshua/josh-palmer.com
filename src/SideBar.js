import React, { Component } from 'react';
import {Link} from 'react-router-dom';

class NavItem extends Component {
    render () {
        let linkStyle = {display: 'block', height: '100%', padding: 10};
        let itemStyle = {width: '100%', background: '#f0f0f0', textAlign: 'center'};
        return (
            <li style={itemStyle} className="navItem">
                {this.props.external
                ? <a href={this.props.path}
                     target="_blank"
                     style={linkStyle}>{this.props.text}</a>
                : <Link to={this.props.path}
                        style={linkStyle}>{this.props.text}</Link>
                }
            </li>
        );
    }
}

class SideBar extends Component {
    render () {
        return (
            <aside className="sideBar">
                <ul style={{listStyleType: 'none', padding: 0}}>
                    {this.props.routes.map((route, index) => (
                        <NavItem key={index} path={route.path} text={route.navText} external={route.external}/>
                    ))}
                </ul>
            </aside>
        );
    }

}

export default SideBar;
