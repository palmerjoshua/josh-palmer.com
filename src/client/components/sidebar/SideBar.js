import React, { Component } from 'react';
import {Link} from 'react-router-dom';

class NavItem extends Component {
    render () {
        let linkStyle = {display: 'block', height: '100%', padding: 10};
        return (
            <li className="navItem">
                {this.props.external
                    ? <a href={this.props.path}
                         target="_blank"
                         style={linkStyle}>{this.props.text}</a>
                    : <Link to={this.props.path}
                            style={linkStyle}>{this.props.text}</Link>
                }
            </li>
        );
    }
}

export default class SideBar extends Component {
    render () {
        return (
            <nav  style={{width: '100%'}} className="sideBar">
                <ul style={{listStyleType: 'none', padding: 0}}>
                    {this.props.routes.filter((route) => !route.hide).map((route, index) => (
                        <NavItem key={index}
                                 path={route.path}
                                 text={route.text}
                                 external={!route.hasOwnProperty('component')}/>
                    ))}
                </ul>
            </nav>
        );
    }
}
