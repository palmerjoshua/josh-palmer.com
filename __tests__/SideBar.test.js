import React from 'react';
import renderer from 'react-test-renderer';

import {BrowserRouter} from 'react-router-dom';
import SideBar from '../src/client/components/sidebar/SideBar';
jest.mock('../src/client/routes/config.js');

it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter routes={require('../src/client/routes/Routes')}><SideBar/></BrowserRouter>);
    expect(tree).toMatchSnapshot();
});
