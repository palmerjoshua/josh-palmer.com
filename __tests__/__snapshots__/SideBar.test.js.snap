exports[`test renders correctly 1`] = `
<div>
  <aside
    className="sideBar">
    <nav
      className="Navigator">
      <ul
        style={
          Object {
            "listStyleType": "none",
            "padding": 0,
          }
        }>
        <li
          className="navItem">
          <a
            href="/"
            style={
              Object {
                "display": "inline-block",
                "height": "100%",
                "padding": 10,
                "width": "90%",
              }
            }>
            Home
          </a>
        </li>
        <li
          className="navItem">
          <a
            href="/bar"
            style={
              Object {
                "display": "inline-block",
                "height": "100%",
                "padding": 10,
                "width": "90%",
              }
            }>
            Bar
          </a>
        </li>
        <li
          className="navItem">
          <a
            href="https://example.com"
            style={
              Object {
                "display": "inline-block",
                "height": "100%",
                "padding": 10,
                "width": "90%",
              }
            }
            target="_blank">
            External
          </a>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;
